src/twisted/application/internet.py

```diff
@@ -1029,13 +1058,31 @@ def __init__(self, endpoint, factory, retryPolicy=None, clock=None):
             this attribute will not be serialized, and the default value (the
             reactor) will be restored when deserialized.
         @type clock: L{IReactorTime}
+
+        @param prepareConnection: A single argument L{callable} that may return
+            a L{Deferred}. It will be called once with the L{protocol
```

Comment:
Some non-blocking, but useful stuff:

- What should this `Deferred` fire with?
- If it can be anything (which is implied) will its callback type be preserved?
- Will errors be handled?  Logged?